### PR TITLE
php files, js scripting, and front ends appear to be all lined up. Mi…

### PR DESCRIPTION
**Pull request by cgrady3**
…ssing portions of js have been identified

### CODE REVIEW

File: LAMP/AddContact.php [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
-		returnWithError($stmt->affected_rows);
-		$stmt->close();
-		$conn->close();
-
-	}
+	// include database connection file
+	include_once "dbConfig.php";
+		
+	$stmt = $conn->prepare("INSERT INTO Contacts (FirstName, LastName, Email, Phone, UserID) VALUES (?, ?, ?, ?, ?)");
+	$stmt->bind_param("ssssi", $inData["FirstName", "LastName", "Email", "Phone", "UserID"]);
```
Comment (kingcolston) [inline LAMP/AddContact.php]: This gives us a 500 Server Error so lets just leave It as this
```suggestion
		$stmt->bind_param("ssssi", $inData["FirstName"], $inData["LastName"], $inData["Email"], $inData["Phone"], $inData["UserID"]);
```

File: LAMP/DeleteContact.php [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-		$stmt->execute();
-		$stmt->close();
-		$conn->close();
-		returnWithError("");
-	}
+	// include database connection file
+	include_once "dbConfig.php";
+
+	$stmt = $conn->prepare("DELETE FROM Contacts WHERE UserID =? AND ContactID =?");
+	$stmt->bind_param("ss", $inData["UserID", "ContactID"]);
```
Comment (kingcolston) [inline LAMP/DeleteContact.php]:
```suggestion
	$stmt->bind_param("ss", $inData["UserID"], $inData["ContactID"]);
```

File: LAMP/DisplayContact.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+	$inData = getRequestInfo();
+
+	$searchResult = "";
+
+	// include database connection file
+	include_once "dbConfig.php";
+
+	$stmt = $conn->prepare("SELECT * FROM Contacts WHERE ContactID=? AND UserID=?");
+	$stmt->bind_param("ss", $inData["ContactID", "UserID"]);
```
Comment (kingcolston) [inline LAMP/DisplayContact.php]:
```suggestion
	$stmt->bind_param("ss", $inData["ContactID"], $inData["UserID"]);
```

File: LAMP/Login.php [hunk 4]
```diff
@@ -7,21 +6,19 @@
...
-	}
-	else
+	// include database connection file
+	include_once "dbConfig.php";
+	
+	function loginUser()
 	{
-		$stmt = $conn->prepare("SELECT UserID,firstName,lastName FROM Users WHERE Email=? AND Password =?");
-		$stmt->bind_param("ss", $inData["email"], $inData["password"]);
+		$stmt = $conn->prepare("SELECT UserID,firstName,lastName FROM Users WHERE Login=? AND Password =?");
```
Comment (kingcolston) [inline LAMP/Login.php]: In the Users table we actually have It as Email
```suggestion
		$stmt = $conn->prepare("SELECT UserID,firstName,lastName FROM Users WHERE Email=? AND Password =?");
```

File: LAMP/Login.php [hunk 5]
```diff
@@ -7,21 +6,19 @@
...
+		$stmt->bind_param("ss", $inData["login"], $inData["password"]);
```
Comment (kingcolston) [inline LAMP/Login.php]:
```suggestion
		$stmt->bind_param("ss", $inData["Email"], $inData["password"]);
```

File: LAMP/Login.php [hunk 6]
```diff
@@ -7,21 +6,19 @@
 	$firstName = "";
 	$lastName = "";
```
Comment (kingcolston) [inline LAMP/Login.php]: We can delete this since its unused

File: LAMP/UpdateContact.php [hunk 7]
```diff
@@ -1,21 +1,15 @@
...
-		returnWithError($stmt->affected_rows);
-		$stmt->close();
-		$conn->close();
-
-	}
+	// include database connection file
+	include_once "dbConfig.php";
+		
+	$stmt = $conn->prepare("UPDATE Contacts SET FirstName=?, LastName=?, Email=?, Phone=? WHERE ContactID=? AND UserID=?");
+	$stmt->bind_param("ssssss", $inData["FirstName", "LastName", "Email", "Phone", "ContactID", "UserID"]);
```
Comment (kingcolston) [inline LAMP/UpdateContact.php]:
```suggestion
	$stmt->bind_param("ssssss", $inData["FirstName"], $inData["LastName"], $inData["Email"], $inData["Phone"], $inData["ContactID"], $inData["UserID"]);
```

**Review by kingcolston — CHANGES_REQUESTED** [review]
After this is changed I can test the search endpoint

File: API/DisplayContact.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+	include "dbConfig.php";
+	include "returnFunctions.php";
+
+	$inData = getRequestInfo();
+
+	$stmt = $conn->prepare("SELECT * FROM Contacts WHERE ContactID=? AND UserID=?");
+	$stmt->bind_param("ss", $inData["ContactID"], $inData["UserID"]);
+	$stmt->execute();
+
+	returnWithInfo($stmt->get_result());
```
Comment (kingcolston) [inline API/DisplayContact.php]:
```suggestion
      $row = $stmt->get_result();
	returnWithInfo($row->fetch_assoc() );
```

File: API/SearchContacts.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ 
+	$stmt = $conn->prepare("SELECT * FROM Contacts WHERE FirstName LIKE ? OR LastName LIKE ? AND UserID=?");
+	$name = $inData["search"] . "%";
+	$stmt->bind_param("sss", $name, $name, $inData["UserID"]);
+	$stmt->execute();
+
+	$result = $stmt->get_result();		
+
+  	if ($result->num_rows > 0) {
+		returnWithInfo($result);
```
Comment (kingcolston) [inline API/SearchContacts.php]:
```suggestion
		returnWithInfo($result->fetch_assoc());
```

File: API/AddContact.php [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<?php
+	// include database connection file
+	include "dbConfig.php";
+	include "returnFunctions.php";
+
+	$inData = getRequestInfo();
+	
+	$stmt = $conn->prepare("SELECT * FROM Contacts WHERE FirstName=?, LastName=?, Email=?, Phone=?, UserID=?");
```
Comment (kingcolston) [inline API/AddContact.php]:
```suggestion
	$stmt = $conn->prepare("SELECT * FROM Contacts WHERE FirstName=? AND LastName=? AND Email=? AND Phone=? AND UserID=?");
```

Comment (kingcolston) [reply API/AddContact.php]: don't ask how long It took me to figure out this was the issue

File: API/AddContact.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+	$stmt->bind_param("ssssi", $inData["FirstName"], $inData["LastName"], $inData["Email"], $inData["Phone"], $inData["UserID"]);
+	$stmt->execute();
+
+	$result = $stmt->get_result();
+
+	if ($result->num_rows > 0){
```
Comment (kingcolston) [inline API/AddContact.php:14]:
```suggestion
	if ($stmt->affected_rows > 0){
```

**Review by kingcolston — CHANGES_REQUESTED** [review]

File: API/AddContact.php (hunk 11 above)
Comment (cgrady3) [reply API/AddContact.php:14]: Line 14 is checking if there already exits that contact so it needs to check if num rows returned from the search is > 0, the else statement below it inits the stmt to be an insert

**Review by kingcolston — APPROVED** [review]

**Pull request merged**